### PR TITLE
Add com.snowplowanalytics.snowplow.r2f/removal_criteria/jsonschema/1-0-0

### PR DESCRIPTION
**Pull request by knservis**
Number of issues to resolve by the reviewers:

- [x] The list of acceptable pojo fields has simply been copied from PII, but do they all make sense for R2F or could they lead to removing data for multiple people?
- [x] Is r2f an acceptable name?
- [x] Is r2f in an acceptable namespace (com.snowplowanalytocs.snowplow)?

### CODE REVIEW

**Comment by alexanderdean** [comment]
Let's keep the naming discussion on the other thread.

I think the list of acceptable pojo fields is a really interesting question. Interested in @yalisassoon's thoughts here.

File: jsonpaths/com.snowplowanalytics.snowplow/r2f_data/jsonschema/1-0-0 [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+          ],
+          "description": "The name of the field"
+        },
+        "jsonPath": {
+          "type": "string",
+          "description": "The JsonPath that was specified"
+        },
+        "valueForWhichRowWillBeDeleted": {
+          "type": "string",
+          "description": "The value that is present in a row ofr that field, will cause it to be deleted"
```
Comment (chuwy) [inline jsonpaths/com.snowplowanalytics.snowplow/r2f_data/jsonschema/1-0-0]: `ofr` -> `row`

File: jsonpaths/com.snowplowanalytics.snowplow/r2f_data/jsonschema/1-0-0 [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            "se_property",
+            "mkt_clickid",
+            "refr_domain_userid",
+            "domain_sessionid"
+          ],
+          "description": "The name of the field"
+        },
+        "valueForWhichRowWillBeDeleted": {
+          "type": "string",
+          "description": "The value that is present in a row ofr that field, will cause it to be deleted"
```
Comment (chuwy) [inline jsonpaths/com.snowplowanalytics.snowplow/r2f_data/jsonschema/1-0-0]: `ofr` -> `for`

File: jsonpaths/com.snowplowanalytics.snowplow/r2f_data/jsonschema/1-0-0 [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            "se_action",
+            "se_label",
+            "se_property",
+            "mkt_clickid",
+            "refr_domain_userid",
+            "domain_sessionid"
+          ],
+          "description": "The name of the field"
+        },
+        "valueForWhichRowWillBeDeleted": {
```
Comment (chuwy) [inline jsonpaths/com.snowplowanalytics.snowplow/r2f_data/jsonschema/1-0-0]: Any chance for better name? `valueToDelete` or just `value` having that we don't call `filedName` a `fieldWhereValueForWhichRowWillBeDeletedShouldBeFound`?

**Comment by knservis** [comment]
@chuwy Thanks for that:

In regards to:

> Also, I'm sorry if it was discussed anywhere already, but shouldn't it contain array of fieldName/value pairs? Running a job just to delete one value sounds wasteful for me.

There will be a list of them. This is a schema on each line of the list.

File: jsonpaths/com.snowplowanalytics.snowplow/r2f_data/jsonschema/1-0-0 (hunk 3 above)
Comment (knservis) [reply jsonpaths/com.snowplowanalytics.snowplow/r2f_data/jsonschema/1-0-0]: Please suggest a better name that both:

* Does state intent.
* Does not mislead the user into believing that the value alone, instead of the entire row will be deleted.

**Comment by alexanderdean** [comment]
Actually @knservis @chuwy I think this is a non-issue and we don't need to wait on Yali - for two reasons:

1. We have the dead man's handle which should stop catastrophic event deletion through a poorly configured forget file
2. Reduce to the absurd, all properties will eventually move out of atomic.events into schemas (this is the atomic.events refactor), and therefore worrying about guarding certain fields in the POJO for now, is not time well-spent (as those fields will all ultimately be up-for-grabs in the future)

TL;DR we can proceed with the current list of fields.

**Comment by knservis** [comment]
Thanks @alexanderdean. In which case @alexanderdean @BenFradet and @chuwy is there something else remaining in this PR?

File: schemas/com.snowplowanalytics.snowplow.r2f/removal_criteria/jsonschema/1-0-0 [hunk 4]
```diff
@@ -0,0 +1,98 @@
+{
+  "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+  "description": "Schema for R2F (Right to be forgotten) criteria files that contain the identifiers for events that should be deleted.",
```
Comment (BenFradet) [inline schemas/com.snowplowanalytics.snowplow.r2f/removal_criteria/jsonschema/1-0-0]: should we specify that it's for only one criteria, reading it it made me think it was the schema for the file.

Comment (alexanderdean) [reply schemas/com.snowplowanalytics.snowplow.r2f/removal_criteria/jsonschema/1-0-0]: Let's change:

> Schema for R2F (Right to be forgotten) criteria files that contain the identifiers for events that should be deleted.

to:

> Schema for an individual entry in an R2F (Right to be forgotten) criteria file, each entry specifying an identifier for events that should be deleted.
